cpp/src/strings/replace/backref_re.cuh

```diff
@@ -61,14 +61,15 @@ struct backrefs_fn {
     auto out_ptr      = d_chars ? (d_chars + d_offsets[idx]) : nullptr;
     size_type lpos    = 0;                   // last byte position processed in d_str
     size_type begin   = 0;                   // first character position matching regex
-    size_type end     = nchars;              // last character position (exclusive)
+    size_type end     = -1;                  // last character position (exclusive)
 
     // copy input to output replacing strings as we go
-    while (prog.find(prog_idx, d_str, begin, end) > 0)  // inits the begin/end vars
+    while ((begin <= nchars) &&
+           (prog.find(prog_idx, d_str, begin, end) > 0))  // inits the begin/end vars
     {
-      auto spos = d_str.byte_offset(begin);             // get offset for the
-      auto epos = d_str.byte_offset(end);               // character position values;
-      nbytes += d_repl.size_bytes() - (epos - spos);    // compute the output size
+      auto spos = d_str.byte_offset(begin);               // get offset for the
+      auto epos = d_str.byte_offset(end);                 // character position values;
```

Comment:
Similarly, does this works correctly with `end = -1`?

Reply:
Yes.